gpslogger/src/main/java/com/mendhak/gpslogger/common/Systems.java

```diff
@@ -234,4 +240,30 @@ else if(appThemeSetting.equalsIgnoreCase("light")){
 
     }
 
+    /**
+     * Returns a OneTimeWorkRequest with the given worker class and data map. The constraints are set to
+     * UNMETERED network type if the user has set the app to only send on wifi. Otherwise it is set to
+     * CONNECTED. The initial delay is set to 1 second to avoid the work being enqueued immediately.
+     * The backoff criteria is set to exponential with a 30 second initial delay.
+     * @param workerClass
+     * @param dataMap
+     * @return
+     */
+    public static OneTimeWorkRequest getBasicOneTimeWorkRequest(Class workerClass, HashMap<String, Object> dataMap) {
```

Comment:
Should be a decent reusable way of building work requests and keeping common options. Especially around the networking and the datamap and the backoff and retry.